File: src/main/java/fi/finlit/edith/domain/UrlElement.java
/**
 *
 */
package fi.finlit.edith.domain;


public class UrlElement implements ParagraphElement {

    private String string;

    private String url;

    public UrlElement() {
    }

    public UrlElement(String string) {
        this.string = string;
    }

    @Override
    public ParagraphElement copy() {
        UrlElement element = new UrlElement(string);
        element.setUrl(url);
        return element;
    }

    @Override
    public String toString() {
        return "<a" + (url == null ? "" : " href=\"" + url + "\"") + ">" + string + "</a>";
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getUrl() {
        return url;
    }

    public String getString() {
        return string;
    }
}

File: src/main/scala/fi/finlit/edith/domain/UrlElement.scala
package fi.finlit.edith.domain

import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

class UrlElement(@BeanProperty val string: String) extends ParagraphElement {

  @BeanProperty
  var url: String = _

  override def copy(): ParagraphElement = {
    var element = new UrlElement(string)
    element.setUrl(url)
    element
  }

  override def toString(): String = {
    "<a" + (if (url == null) "" else " href=\"" + url + "\"") + ">" + string + "</a>"
  }
}
